Add deprecation section to Coding Guide

pjcozzi opened the pull request:
This just moves the old Deprecation Guide (with minor improvements) to a section of the Coding Guide:

https://github.com/AnalyticalGraphicsInc/cesium/blob/deprecation/Documentation/Contributors/CodingGuide/README.md#deprecation-and-breaking-changes

mramato commented:
We should mention that all deprecated usage should be removed from the code base (except for specs that specifically test the deprecated functionality).

Other than that, I'm okay with this for now, but it should probably detail the different use cases for deprecation in the future.  For example, we occasionally deprecate only a specific parameter or deprecate because of a rename. In these cases we add new tests to cover both the deprecated and non-deprecated behavior and things can get a bit more involved than simply deprecating a function not used anywhere else.

pjcozzi commented:
> We should mention that all deprecated usage should be removed from the code base (except for specs that specifically test the deprecated functionality).

Good call.  Updated.

> Other than that, I'm okay with this for now, but it should probably detail the different use cases for deprecation in the future. For example, we occasionally deprecate only a specific parameter or deprecate because of a rename. In these cases we add new tests to cover both the deprecated and non-deprecated behavior and things can get a bit more involved than simply deprecating a function not used anywhere else.

Updated #3348.
